src/startupscriptgenerator/dotnetcore/scriptgenerator.go

```diff
@@ -24,59 +20,52 @@ type DotnetCoreStartupScriptGenerator struct {
 	Manifest           common.BuildManifest
 }
 
-type projectDetails struct {
-	Name         string
-	FullPath     string
-	Directory    string
-	CSProjectObj csProject
-}
-
-// Object models representing .NET Core '.csproj' file xml content
-type csProject struct {
-	SdkName    string          `xml:"Project,Sdk"`
-	Properties []propertyGroup `xml:"PropertyGroup"`
-}
-
-type propertyGroup struct {
-	AssemblyName string `xml:"AssemblyName"`
-}
-
-const ProjectEnvironmentVariableName = "PROJECT"
 const DefaultBindPort = "8080"
-
-var _projDetails projectDetails = projectDetails{}
+const RuntimeConfigJsonExtension = "runtimeconfig.json"
 
 func (gen *DotnetCoreStartupScriptGenerator) GenerateEntrypointScript(scriptBuilder *strings.Builder) string {
 	logger := common.GetLogger("dotnetcore.scriptgenerator.GenerateEntrypointScript")
 	defer logger.Shutdown()
 
-	logger.LogInformation(
-		"Generating script for source at '%s' and published output at '%s'",
-		gen.SourcePath,
-		gen.AppPath)
-
-	startupDllFileName := gen.getStartupDllFileName()
+	logger.LogInformation("Generating script for published output at '%s'", gen.AppPath)
 
 	// Expose the port so that a custom command can use it if needed
 	common.SetEnvironmentVariableInScript(scriptBuilder, "PORT", gen.BindPort, DefaultBindPort)
 	scriptBuilder.WriteString("export ASPNETCORE_URLS=http://*:$PORT\n\n")
 
+	defaultAppFileDir := filepath.Dir(gen.DefaultAppFilePath)
+
 	scriptBuilder.WriteString("readonly appPath=\"" + gen.RunFromPath + "\"\n")
 	scriptBuilder.WriteString("userStartUpCommand=\"" + gen.UserStartupCommand + "\"\n")
 	scriptBuilder.WriteString("startUpCommand=\"\"\n")
-	scriptBuilder.WriteString("readonly startupDllFileName=\"" + startupDllFileName + "\"\n")
+	scriptBuilder.WriteString("readonly defaultAppFileDir=\"" + defaultAppFileDir + "\"\n")
 	scriptBuilder.WriteString("readonly defaultAppFilePath=\"" + gen.DefaultAppFilePath + "\"\n")
+	scriptBuilder.WriteString("echo Finding the startup file name...\n")
+	scriptBuilder.WriteString("cd \"$appPath\"\n")
+	scriptBuilder.WriteString("for file in *; do \n")
+	scriptBuilder.WriteString("  if [ -f \"$file\" ]; then \n")
```

Comment:
shouldn't we first fallback to the previous way, looking for the .csproj file, before only having this? I'm thinking that if we find any issue with this approach in prod, we will have no alternatives

Reply:
I have tested this with all versions of SDK and this approach is totally fine.